x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1149,22 +1150,48 @@ private static LogicalPlan skipPlan(UnaryPlan plan) {
         @Override
         protected LogicalPlan rule(LogicalPlan plan) {
             Holder<LocalRelation> optimizedPlan = new Holder<>();
-            plan.forEachDown(Project.class, p -> {
-                List<Object> values = extractConstants(p.projections());
-                if (values.size() == p.projections().size() && (p.child() instanceof EsRelation) == false &&
-                    isNotQueryWithFromClauseAndFilterFoldedToFalse(p)) {
+            var leafRelation = leafRelation(plan);
```

Comment:
Personally, I'm not fun of `var` in the code, I like more in the tests for simplification, but let's wait for the opinion of the others.

Reply:
Personally I also find that making the type explicit is more readable, although there are places in ES - less so in *QL, I think - where `var` is used.

Reply:
👍 Costin also pointed out that `var` is not yet available in Java 8, so not using it also helps a lot with backporting the fix to 7.x.